### PR TITLE
Improving file tree building for markdown import

### LINKED ISSUE #9951: MooseDocs markdown import needs to be smarter
## Rationale

As we add documentation that is included in other applications the ability to grab markdown content 
from the framework must be flexible to allow for entire modules to be included without getting um-wanted files.

## Description

You can reproduce the problem by adding a new file to MOOSE, if the file is not included in the app it breaks.

## Impact

Improving the import should lessen the burden for keeping include/exclude lists up-to-date.

### PR DESCRIPTION
**Pull request by aeslaughter**
(closes #9951)

### CODE REVIEW

**Comment by aeslaughter** [comment]
I still need to work on integrating this with the apps, this is just to see how far I got w/r/t MOOSE.

**Comment by aeslaughter** [comment]
This is ready, I have patches waiting for applications.

File: python/MooseDocs/common/moose_docs_import.py [hunk 1]
```diff
@@ -14,11 +14,30 @@
...
 import logging
 import MooseDocs
 LOG = logging.getLogger(__name__)
 
+def build_regex(pattern):
+    """
+    Build regex from paths with * and **.
+    """
+    out = pattern.replace('.', r'\.')
+    out = re.sub(r'(?!<^|/)(\*{2})(?=/|$)', r'(?:.*?)', out, flags=re.MULTILINE)
```
Comment (permcody) [inline python/MooseDocs/common/moose_docs_import.py]: I really think you should document these. This is pretty complex and will be difficult to maintain later. You can be pretty extreme with how you break them up: 
https://github.com/idaholab/moose/blob/devel/modules/chemical_reactions/src/actions/AddCoupledEqSpeciesAction.C#L51-L59

**Review by permcody — CHANGES_REQUESTED** [review]
I'd like to see some docs on the complex regexes you are using. This method doesn't even say what it's doing anywhere.

**Review by brianmoose — APPROVED** [review]

**Review by permcody — APPROVED** [review]

**Pull request merged**